spec-tests/src/test/java/org/candlepin/spec/ExportSpecTest.java

```diff
@@ -0,0 +1,649 @@
+    private static final String CONSUMER_TYPE_PATH = "export/consumer_types/";
+    private static final String CDN_PATH = "export/content_delivery_network/";
+    private static final String RULES_PATH = "export/rules2/";
+
+    @Test
+    @DisplayName("should allow manifest creation with read only user")
+    void shouldAllowManifestCreationWithReadOnlyUser() throws Exception {
+        ApiClient client = ApiClients.admin();
+        OwnerDTO owner = client.owners().createOwner(Owners.random());
+        ConsumerDTO readOnlyConsumer = createReadOnlyConsumer(client, owner);
+
+        File readOnlyManifest = createExport(client, readOnlyConsumer.getUuid(), null);
+        ZipFile readOnlyExport = ExportUtil.getExportArchive(readOnlyManifest);
+        String path = EXPORT_PATH + "consumer.json";
+
+        ConsumerDTO actualConsumer = ExportUtil
+            .deserializeJsonFile(readOnlyExport, path, ConsumerDTO.class);
+        assertEquals(readOnlyConsumer.getUuid(), actualConsumer.getUuid());
+        assertEquals(readOnlyConsumer.getName(), actualConsumer.getName());
+
+        assertEquals(owner.getId(), actualConsumer.getOwner().getId());
+        assertEquals(owner.getKey(), actualConsumer.getOwner().getKey());
+        assertEquals(owner.getDisplayName(), actualConsumer.getOwner().getDisplayName());
+    }
+
+    @Nested
+    @DisplayName("Standard Export Tests")
+    @TestInstance(Lifecycle.PER_CLASS)
+    class StandardExporter {
+        private ApiClient client;
+        private OwnerClient ownerApi;
+        private OwnerContentApi ownerContentApi;
+        private OwnerProductApi ownerProductApi;
+        private CdnApi cdnApi;
+        private ConsumerClient consumerApi;
+        private ConsumerTypeApi consumerTypeApi;
+        private RulesApi rulesApi;
+        private RolesApi rolesApi;
+        private UsersApi usersApi;
+        private JobsClient jobsApi;
+
+        private OwnerDTO owner;
+        private ConsumerDTO consumer;
+        private CdnDTO cdn;
+        private ZipFile export;
+        private File manifest;
+        private Map<String, ProductDTO> productIdToProduct = new HashMap<>();
+
+        @BeforeAll
+        void beforeAll() throws Exception {
+            client = ApiClients.admin();
+            ownerApi = client.owners();
+            ownerContentApi = client.ownerContent();
+            ownerProductApi = client.ownerProducts();
+            cdnApi = client.cdns();
+            consumerApi = client.consumers();
+            consumerTypeApi = client.consumerTypes();
+            rulesApi = client.rules();
+            rolesApi = client.roles();
+            usersApi = client.users();
+            jobsApi = client.jobs();
+
+            initializeData();
+
+            manifest = createExport(client, consumer.getUuid(), cdn);
+            export = ExportUtil.getExportArchive(manifest);
+        }
+
+        @AfterAll
+        void afterAll() throws Exception {
+            export.close();
```

Comment:
Make sure to check that this is null in the event an exception is thrown in `createExport`.

Reply:
Good catch! Updated